snap/info.go

```diff
@@ -358,6 +358,8 @@ type Info struct {
 
 	// OriginalLinks is a map links keys to link lists
 	OriginalLinks map[string][]string
+
+	Categories CategoryInfos
```

Comment:
I'd drop the `CategoryInfos` type and make the fact this is a slice here explicit:

```suggestion
	Categories []CategoryInfo
```

The only case where a similar type is introduced is `MediaInfos`, but in that case a method is added to the slice. It probably isn't worth doing here.